e2e/k8s_test.go

```diff
@@ -132,3 +152,238 @@ func TestLargeDecodedVariableSecretProvidedK8s(t *testing.T) {
 
 	testenv.Test(t, f.Feature())
 }
+
+func TestMultiplePodsChangingPwdInbetweenSecretProvidedK8s(t *testing.T) {
+	f := features.New("multiple pods changing pwd inbetween").
+		// Replaces TEST_ID_2_multiple_pods_changing_pwd_inbetween
+		Assess("secret set correctly in pod1", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			var stdout, stderr bytes.Buffer
+			command := []string{"printenv", "|", "grep", "TEST_SECRET"}
+
+			RunCommandInSecretsProviderPod(cfg.Client(), command, &stdout, &stderr)
+
+			assert.Contains(t, stdout.String(), "supersecret")
+
+			return ctx
+		}).
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			platform := os.Getenv("PLATFORM")
+
+			if platform == "kubernetes" {
+				err := SetConjurSecret(cfg.Client(), "secrets/test_secret", "secret2")
+				if err != nil {
+					fmt.Errorf("error setting conjur secret: %s", err)
+				}
+			} else if platform == "openshift" {
```

Comment:
I would first check if either the kubernetes Go API or the e2e-framework library includes any functionality to do what you need here. It looks like we need two operations:
- delete pods
- scale deployments